pkg/cli/options.go

```diff
@@ -152,6 +153,18 @@ func WithCompletion() Option {
 	}
 }
 
+// WithFilesystem is an Option that allows to set the filesystem used in the CLI.
+func WithFilesystem(fs machinery.Filesystem) Option {
+	return func(c *CLI) error {
+		if fs.FS == nil {
+			return errors.New("invalid filesystem")
+		}
+
+		c.fs = fs
+		return nil
+	}
+}
```

Comment:
I am OK with we create this one.
Just one thing, see that you can pass the FS such as

```
&CLI{
   fs: machinery.Filesystem{FS: afero.NewMemMapFs()},
}
```

as well.